lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -131,6 +131,12 @@
     description:
       - Repository path and tag used to create the container. If an image is not found or pull is true, the image
         will be pulled from the registry. If no tag is included, 'latest' will be used.
+  init:
+    description:
+      - Run an init inside the container that forwards signals and reaps processes.
+        This option requires Docker API 1.25+.
```

Comment:
Please provide input sanity checking against the docker API version so that if an user attempts to use `init` and doesn't have a new enough version of the API, they don't get a stack trace but instead a proper ansible `fail_json` message.

Reply:
@maxamillion 
Thank you for your review.
I have added the validation of docker API's version.

Reply:
@maxamillion @suzuki-shunsuke The validation check wasn't enough, as `init` is still passed on, which leads to an error (see #42162).